File: arc019/b.cpp
#include<string>
#include<iostream>
using namespace std;

int main()
{
  string s;
  cin >> s;
  if(s.length() == 1)
  {
    cout << 0 << endl;
    return 0;
  }
  cout << s.length() << endl;
  int not_symmetry_count = 0;
  for(int i = 0; i < s.length() / 2; i++)
  {
    if(s[i] != s[s.length() - i - 1])
      not_symmetry_count++;
  }
  
  if(not_symmetry_count != 1)
    cout << 25 * s.length() << endl;
  else
    cout << 25 * s.length() - 2 << endl;
}

File: arc019/b.py
s = input()
if len(s) == 1:
  print(0)
  exit()

not_symmetry_count = 0
for i in range(len(s) // 2):
  if s[i] != s[-i - 1]:
    not_symmetry_count += 1

if not_symmetry_count == 0 and len(s) % 2 == 1:
  print(25 * (len(s) - 1))
elif not_symmetry_count != 1:
  print(25 * len(s))
else:
  print(25 * len(s) - 2)
